composer.json

```diff
@@ -12,40 +12,38 @@
     "require" : {
         "php": "^5.6 || ^7.0",
         "cocur/slugify": "^1.4 || ^2.5",
-        "friendsofsymfony/comment-bundle": "^2.0.13",
         "friendsofsymfony/rest-bundle": "^1.8 || ^2.2",
-        "friendsofsymfony/user-bundle": "^1.3",
```

Comment:
This bundle is used in the code, maybe `^1.3 || ^2.0`?

Reply:
Done

Reply:
It depends: Your code is using fosUser? then you habe to depend on it. Why? because dropping a dependency is not a BC break. We might drop FosUser from SonataUser and this bundle would break then. AFAIK we are using FosUser classes here

Reply:
It's only `FOS\UserBundle\Model\UserInterface` that is used in `CustomerInterface`, `CustomerSelector`, `BaseCustomer`.
I think, we should replace it with `Sonata\UserBundle\Model\UserInterface` and fully rely on SonataUserBundle.

Reply:
And why dont we use directly userinterface from symfony?

Reply:
May be. I dont know why this specific check for `FOS\UserBundle\Model\UserInterface` was added.
`CustomerSelector`
```php
       if (true === $this->securityContext->isGranted('IS_AUTHENTICATED_FULLY')) {
            // user is authenticated
            $user = $this->securityContext->getToken()->getUser();

            if (!$user instanceof UserInterface) {
                throw new \RuntimeException('User must be an instance of FOS\UserBundle\Model\UserInterface');
            }

            $customer = $this->customerManager->findOneBy([
                'user' => $user->getId(),
            ]);
        }
```

If we can safely replace it with `Symfony\Component\Security\Core\User\UserInterface` than we definitely should do it.

Reply:
Can you create a small PR with this replacement and drop user-bundle there?

Reply:
Yes, but replace with `Sonata\UserBundle\Model\UserInterface` or `Symfony\Component\Security\Core\User\UserInterface` ?

I noticed that `Sonata\UserBundle\Model\UserInterface` is used in `BaseInvoice` and some other files. If someone is using checks like above, it will break BC.
So I think it should be `Sonata\UserBundle\Model\UserInterface`, at least for now.

Reply:
Changing a type hint to a parent is not a bc break. Thats why it is better to use symfony userinterface. Is the grandparent of all users

Reply:
@jordisala1991 #490 check it, please